app/models/grant_submission.rb

```diff
@@ -1,19 +1,28 @@
 class GrantSubmission < ActiveRecord::Base
+  belongs_to :grant
+  belongs_to :artist
+
+  has_many :proposals
+  has_many :votes
+  has_many :voter_submission_assignments
+
+  delegate :max_funding_dollars, to: :grant
+
   mount_uploader :proposal, GrantProposalUploader
 
-  validates :name, :presence => true, length: { minimum: 4 }
+  validates :name, presence: true, length: { minimum: 4 }
   # Can't require proposal because modification might not change the proposal.
   # validates :proposal, :presence => true
 
-  validates :grant_id, :presence => true
+  validates :grant, presence: true
 
   # Max value depends on the grant, so don't constrain here.
-  validates :requested_funding_dollars, :presence => true, :numericality => {:greater_than => 0, :only_integer => true}
+  validates :requested_funding_dollars, presence: true, numericality: { greater_than: 0, less_than_or_equal_to: :max_funding_dollars, only_integer: true }
```

Comment:
This comment was marked as outdated but is still valid. I just updated the comment above it.